Switch to SQLite3, Upgrade Ruby and Rails, Remove Unused Gems, and Add Fly.io Deployment Configuration

cirdes opened the pull request:
New server: https://rbui.fly.dev

This PR includes the following updates and changes:

1. **Database Switch**:
   - Removed PostgreSQL and Redis, and switched to SQLite3 for simpler web stack.

2. **Version Upgrades**:
   - Updated Ruby to version 3.3.4.
   - Updated Rails to version 7.2.0.beta3.

3. **Asset Pipeline Update**:
   - Removed Sprockets and added Propshaft for managing assets.

4. **Removed Unused Gems**:
   - Removed the following unused gems:
     - `stripe`
     - `letter_opener` (development group)
     - `postmark-rails` (used in production)
     - `octokit` (version ~> 5.0)
     - `faraday-retry` (version ~> 2.0) - used for API calls failing in the `octokit` gem
     - `rack-www` - used to redirect www to non-www subdomain
     - `rollbar`

5. **Deployment Configuration**:
   - Added configuration for deployment on Fly.io.

These changes streamline the project dependencies, simplify development, and prepare the application for deployment on Fly.io.

iseth commented on .yarn/releases/yarn-1.22.19.cjs:
can we get rid of this?

cirdes replied:
@iseth, I have tested running `fly launch` from the `main` branch here, and it creates all of those files. Maybe we should keep it. 

<img width="604" alt="Screenshot 2024-07-29 at 11 43 19 AM" src="https://github.com/user-attachments/assets/99274936-7181-48e8-be36-426a9fe82388">

iseth commented on .yarnrc:
can we get rid of this?

cirdes replied:
I don't think we should because it was generated by fly

iseth replied:
Yeah i just got things working on my end and saw that too. Should it really be checked in to git though?

iseth commented on Gemfile:
```diff
@@ -1,37 +1,30 @@
 
-# Build JSON APIs with ease [https://github.com/rails/jbuilder]
-gem "jbuilder"
+# # Build JSON APIs with ease [https://github.com/rails/jbuilder]
```
this too haha

iseth commented on package.json:
```diff
@@ -2,21 +2,22 @@
-  }
-}
+  },
+  "packageManager": "yarn@1.22.19"
```
this might have something to do with the yarn release files. Look into removing this.

cirdes replied:
This was also added by the Fly CLI. It seems to be a good practice to specify the Yarn version here to ensure that the specific Yarn version is always used.

<img width="1069" alt="Screenshot 2024-07-29 at 11 51 37 AM" src="https://github.com/user-attachments/assets/990af696-a5bd-4ff3-be4e-be8e3ab0f209">

iseth reviewed (changes requested):
almost perfect

cirdes commented:
@iseth, I have removed the ".tool-versions" file and fixed the comment in the Gemfile.

All other issues you mentioned were generated by the fly.io CLI, and I believe they are recommended by Yarn.

Let me know if you need anything else!

iseth commented:
> @iseth, I have removed the ".tool-versions" file and fixed the comment in the Gemfile.
> 
> All other issues you mentioned were generated by the fly.io CLI, and I believe they are recommended by Yarn.
> 
> Let me know if you need anything else!

I dig too deep into this but i think its definetly coming from https://github.com/fly-apps/dockerfile-rails/blob/2711cf5aab3716c3dafd1c08307ee62107a45dda/lib/generators/templates/_npm_install.erb

cirdes commented:
@iseth, I have removed .yarn/* and .yarnrc and added them to .gitignore.

iseth reviewed (approved):
awesome!
